### ISSUE TITLE
Using HTTP proxies errors.

### ISSUE THREAD

**Issue by Birbe** [PrismarineJS/node-minecraft-protocol#649]
I'm trying to use HTTP proxies with NMP, and whenever I use a (tested, functioning) proxy, then I get a "Read error for name : 84 is not in the mappings value". I've tested the account, it works, as if I don't use a proxy then it logs into my test server fine. I've also tested my code, and it isn't wrong, as using the official HTTP proxy example gives the same error.

Error: https://throwbin.io/AHoeIS8
Code: https://throwbin.io/wm9YQ3H

**Comment by rom1504** [comment]
How do you know your proxy works ? Does it support CONNECT method ?
Did you try using a local proxy ?

**Comment by Gumballll** [comment]
I know because I've
A. test the proxies, they work.
B. The only time I don't get a socket based error is this error, an NMP error, which I would have to think the proxy works.

**Comment by Gumballll** [comment]
I might have solved the issue, disregard. You can probably lock the topic

**Comment by Gumballll** [comment]
Nope, no. I didn't solve it. Even when I do get a functioning HTTP proxy NMP just errors.

**Comment by rom1504** [comment]
How did you test that the proxy work ? Does it support CONNECT method ?
Did you try using a local proxy ?

**Comment by Gumballll** [comment]
Yeah, I tested it. I'm running a local server on 1.14.3. The account works fine, as I'm using my main and no proxy works fine. Using a local proxy (which I know for a fact works too) gives me no errors on the proxy server side, but I do get a "connection lost" immediately after connection.

**Comment by Gumballll** [comment]
Okay, so I figured it out. I had to write my own custom proxy server in node.js. It's a really simple one, it literally pipes both connections through, and *this* worked on my local test server, I'll have to test it on a public server but this seems to be working. It appears that every proxy that I've tried, even when tested using online services and given an OK, did not really want to work here. They did seem to support CONNECT, but somehow TCP might be screwed with them. I know that the proxies work one way or another, so I'm just gonna write my own proxy service that pretty much double proxies the connection, maybe with different kinds of proxies.

NMP <-> Custom Proxy Service <-> External Proxies <-> Minecraft Server

Really sorry for bugging you, socket coding is a huge PITA

**Comment by Gumballll** [comment]
Okay, so adding to my above comment, using servers other than my local one gives me an ECONNREFUSED error. Not sure why.

**Comment by rom1504** [comment]
Node-minecraft-protocol allows you to pass any socket, so if you manage to
create a tcp socket with any kind of proxy method, that will work.
So you can integrate your custom proxy thing directly in that, no need to
use another process

**Comment by sav3nwarrioR** [comment]
@Gumballll I'm having the same problem. Could you please share your code here? I tried a lot of things, but nothing works for me. The example (client-socket-proxy) gives me Negotiation Error. I'm just dumb and can't understand anything.

**Comment by rom1504** [comment]
Ok to be clear : there are probably no public http proxy which supports CONNECT.
So just use a socks proxy

**Comment by Gumballll** [comment]
@sav3nwarrioR What rom1504 said is correct, just use a SOCKS proxy.

**Comment by sav3nwarrioR** [comment]
I tried to use NordVPN SOCKS5 proxy from [here](https://nordvpn.com/servers/tools/?technologies=Socks+5) with client_socks_proxy example code. It worked - I got "connected" message in the console but immediately after that "Connection lost" appeared. I have several NordVPN accounts so I tried to switch between accounts but it kept losing connection immediately after connecting. I tried switching servers and locations. After several attempts I got "Error: Negotiation Error (6)" error message. And I was stuck with it: switching servers/locations/acoounts couldn't help.
